### PR TITLE
Feature/rename compound outlets

### PR DESCRIPTION
**Pull request by RyanM-RMA**
Renaming compound outlets to Virtual outlets, per our CDA dev meeting on 7/9/24.
https://www.hec.usace.army.mil/confluence/display/CWMS/2024-07-09+Developer+Meeting

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java [hunk 1]
```diff
@@ -223,18 +207,25 @@ private CompoundOutletRecord mapToCompoundRecord(List<String> table, String offi
         if (!table.isEmpty()) {
             outlet = table.get(0);
         }
-        return new CompoundOutletRecord.Builder().withOutletId(
+        return new VirtualOutletRecord.Builder().withOutletId(
```
Comment (adamkorynta) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java:210]: Missing projectId

File: cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/VirtualOutlet.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import cwms.cda.formatters.json.JsonV1;
+import java.util.ArrayList;
+import java.util.List;
+
+@FormattableWith(contentType = Formats.JSONV1, formatter = JsonV1.class, aliases = {Formats.DEFAULT, Formats.JSON})
+@JsonDeserialize(builder = VirtualOutlet.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+@JsonPropertyOrder({"projectId", "virtualOutletId", "virtualRecords"})
+public class VirtualOutlet extends CwmsDTOBase {
```
Comment (adamkorynta) [inline cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/VirtualOutlet.java]: final

File: cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/VirtualOutlet.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        return projectId;
+    }
+
+    @Override
+    protected void validateInternal(CwmsDTOValidator validator) {
+        super.validateInternal(validator);
+        validator.validateCollection(virtualRecords);
+    }
+
+    public static class Builder {
```
Comment (adamkorynta) [inline cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/VirtualOutlet.java]: final

**Review by adamkorynta — CHANGES_REQUESTED** [review]

File: cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java (hunk 1 above)
Comment (RyanM-RMA) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java:210]: The project id is part of the VirtualOutlet, not the records.

Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java:210]: ah gotcha

File: cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/VirtualOutlet.java (hunk 2 above)
Comment (RyanM-RMA) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/VirtualOutlet.java]: Finalized

File: cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/VirtualOutlet.java (hunk 3 above)
Comment (RyanM-RMA) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/VirtualOutlet.java]: Finalized

**Review by adamkorynta — APPROVED** [review]

**Review by rma-psmorris — COMMENTED** [review]
Post-merge review: accepted, I like the separation of compound outlets to be outside of the common outlet.

**Pull request merged**